### PR TITLE
Locomotor cache should handle custom layers

### LINKED ISSUE #17031: High bridges are broken
Units do not pass over them anymore.

resim

### PR DESCRIPTION
**Pull request by teinarss**
Fixes #17031

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/World/Locomotor.cs [hunk 1]
```diff
@@ -238,15 +241,15 @@ public short MovementCostForCell(CPos cell)
 			if (!world.Map.Contains(cell))
 				return short.MaxValue;
 
-			return cellsCost[cell];
+			return cell.Layer == 0 ? cellsCost[cell] : customLayerCellsCost[cell.Layer][cell];
```
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/World/Locomotor.cs]: Just thinking aloud: Would it maybe make more sense to use the new `customLayer*` caches for `cell.Layer == 0` as well and drop the old `cellsCost` and `blockingCache`? Or would that negatively impact perf on the default layer?

**Comment by teinarss** [comment]
That's what I'm thinking, we have the same pattern in the PathGraph class.

Also I have discovered several flaws in this PR. So wait with further
reviewing until I have updated!

**Comment by pchote** [comment]
Any progress here @teinarss?

**Comment by teinarss** [comment]
Yeah u just need to wrap it up, will do it later today!

**Comment by teinarss** [comment]
Upadated. Added a new interface `IWorldPostLoaded` (Name can be debated) but the purpose is to have a point where we know that the world is completely loaded. In this case the custom movement layers. 

Also im using this interface in my prototype for a fix on #5908

**Comment by pchote** [comment]
> the purpose is to have a point where we know that the world is completely loaded

In previous cases we have handled this either by using `Requires<TInfo>` if we need to be loaded after a specific trait type, queueing a `FrameEndTask` if we need to do something after everything else (that didnt run after us and queue its own FrameEndTask), or both.

**Comment by teinarss** [comment]
Updated

**Review by pchote — APPROVED** [review]

**Review by tovl — APPROVED** [review]
Fixes jumpjets, subterranean units and tunnels as well. :+1:

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/519325b9379b5ac9e4a848e3af9243efcceffb64...b8cbaf4e3e8d956a9fad390ac240a08d23b19b96)

https://github.com/OpenRA/OpenRA/commit/a221bac3e673523a83148d74327a3f2b493693d9

**Pull request merged**